### PR TITLE
Update 0x08-Testing Tools

### PR DESCRIPTION
**Pull request by cpholguera**
Done in this PR:
- Format updated according to issue #1186
- Removed duplicated names of the tools after the colons (:), replaced with "A tool ...", "A framework" ..;
- general format and rewording for consistency;
- several typos fixed;
- Hopper moved to RE and Static Analysis because it can also be used to RE for Android;
- iGoat-Swift added;
- AndBug and Introspy-Android removed, they're not updated since 2014 and only support Android 2.3 - 4.3. This page says that "all tools below have been verified if they are alive". Maybe there are more dead tools.
- Also removed Introspy-Android from 0x05j

We should open an issue to review other dead tools. 

This PR covers issue #1186 and #1196.

### CODE REVIEW

File: Document/0x08-Testing-Tools.md [hunk 1]
```diff
@@ -6,166 +6,163 @@ To perform security testing different tools are available in order to be able to
...
-- [SUPER](https://github.com/SUPERAndroidAnalyzer/super "SUPER") - SUPER is a command-line application that can be used in Windows, macOS and Linux, that analyzes .apk files in search for vulnerabilities.
-- [SpotBugs](https://spotbugs.github.io/ "SpotBugs") - Static Analysis tool for Java
+ - Androguard: A python based tool, which can use to disassemble and decompile Android apps - https://github.com/androguard/androguard
+ - Android Backup Extractor: Utility to extract and repack Android backups created with adb backup (ICS+). Largely based on BackupManagerService.java from AOSP - https://github.com/nelenkov/android-backup-extractor
+ - Android Debug Bridge (adb): A versatile command line tool used to communicate with an emulator instance or connected Android device - https://developer.android.com/studio/command-line/adb.html
+ - APKTool: A tool for reverse engineering 3rd party, closed, binary Android apps. It can decode resources to nearly original form and rebuild them after making some modifications - https://ibotpeaches.github.io/Apktool/
```
Comment (sushi2k) [inline Document/0x08-Testing-Tools.md]: Can you please change to "Apktool"

File: Document/0x08-Testing-Tools.md [hunk 2]
```diff
@@ -6,166 +6,163 @@ To perform security testing different tools are available in order to be able to
...
+ - Qark: A tool designed to look for several security related Android application vulnerabilities, either in source code or packaged APKs - https://github.com/linkedin/qark
+ - Sign: A Java JAR executable (Sign.jar) which automatically signs an apk with the Android test certificate - https://github.com/appium/sign
+ - Simplify: A tool for de-obfuscating android package into Classes.dex which can be use Dex2jar and JD-GUI to extract contents of dex file - https://github.com/CalebFenton/simplify
+ - SUPER: A command-line application that can be used in Windows, macOS and Linux, that analyzes .apk files in search for vulnerabilities - https://github.com/SUPERAndroidAnalyzer/super
+ - SpotBugs: Static analysis tool for Java - https://spotbugs.github.io/
 
 #### Dynamic and Runtime Analysis
 
-- [Android Tcpdump](https://www.androidtcpdump.com "TCPDump") - A command line packet capture utility for Android.
-- [Cydia Substrate: Introspy-Android](https://github.com/iSECPartners/Introspy-Android "Introspy Android") - Blackbox tool to help understand what an Android application is doing at runtime and assist in the identification of potential security issues.
```
Comment (sushi2k) [inline Document/0x08-Testing-Tools.md]: This here was removed but not added. Can you please add?

File: Document/0x08-Testing-Tools.md [hunk 3]
```diff
@@ -6,166 +6,163 @@ To perform security testing different tools are available in order to be able to
...
-- [jdb](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jdb.html) - The jdb tool is a Java Debugger which allows you to set breakpoints and print application variables. Jdb uses the JDWP protocol.
-- [Andbug](https://github.com/swdunlop/AndBug) - A Scriptable Android Debugger; AndBug is a debugger targeting the Android platform's Dalvik virtual machine intended for reverse engineers and developers. andbug is using the Java Debug Wire Protocol (JDWP
-- [Introspy-Android](https://github.com/iSECPartners/Introspy-Android) - Blackbox tool to help understand what an Android application is doing at runtime and assist in the identification of potential security issues. Limitations: Introspy worked based on 'Cydia Substrate', so to work we need the Cydia and based on the app website Cydia, Supported Android on versions 2.3 through 4.3. On the mentioned version the Introspy does working correctly. (http://www.cydiasubstrate.com/)
```
Comment (sushi2k) [inline Document/0x08-Testing-Tools.md]: Can you also please add this one again?

File: Document/0x08-Testing-Tools.md [hunk 4]
```diff
@@ -6,166 +6,163 @@ To perform security testing different tools are available in order to be able to
...
-- [logcat-color](https://github.com/marshall/logcat-color "Logcat color") - A colorful and highly configurable alternative to the adb logcat command from the Android SDK.
-- [VirtualHook](https://github.com/rk700/VirtualHook "VirtualHook") - VirtualHook is a hooking tool for applications on Android ART(>=5.0). It's based on VirtualApp and therefore does not require root permission to inject hooks.
-- [Xposed Framework](https://forum.xda-developers.com/xposed/xposed-installer-versions-changelog-t2714053 "Xposed Framework") - Xposed framework enables you to modify the system or application aspect and behavior at runtime, without modifying any Android application package(APK) or re-flashing.
-- [jdb](https://docs.oracle.com/javase/7/docs/technotes/tools/windows/jdb.html) - The jdb tool is a Java Debugger which allows you to set breakpoints and print application variables. Jdb uses the JDWP protocol.
-- [Andbug](https://github.com/swdunlop/AndBug) - A Scriptable Android Debugger; AndBug is a debugger targeting the Android platform's Dalvik virtual machine intended for reverse engineers and developers. andbug is using the Java Debug Wire Protocol (JDWP
```
Comment (sushi2k) [inline Document/0x08-Testing-Tools.md]: Can you also please add this one again?

File: Document/0x08-Testing-Tools.md [hunk 5]
```diff
@@ -6,166 +6,163 @@ To perform security testing different tools are available in order to be able to
...
-- [DVIA-v2](https://github.com/prateek147/DVIA-v2 "Damn Vulnerable iOS App v2") - A vulnerable iOS app, written in Swift with over 15 vulnerabilities.
-- [iGoat](https://github.com/owasp/igoat "iGoat") - iGoat is a learning tool for iOS developers (iPhone, iPad, etc.) and mobile app pentesters. It was inspired by the WebGoat project, and has a similar conceptual flow to it.
+ - Crackmes: A set of applications to test your iOS application hacking skills - https://github.com/OWASP/owasp-mstg/tree/master/Crackmes
+ - Myriam: A vulnerable iOS app with iOS security challenges - https://github.com/GeoSn0w/Myriam
+ - DVIA: A vulnerable iOS app which provides a platform to mobile security enthusiasts/professionals or students to test their iOS penetration testing skills - http://damnvulnerableiosapp.com/
+ - DVIA-v2: A vulnerable iOS app, written in Swift with over 15 vulnerabilities - https://github.com/prateek147/DVIA-v2
+ - iGoat: An iOS app serving as a learning tool for iOS developers (iPhone, iPad, etc.) and mobile app pentesters. It was inspired by the WebGoat project, and has a similar conceptual flow to it - https://github.com/owasp/igoat
```
Comment (sushi2k) [inline Document/0x08-Testing-Tools.md]: Can you please indicaet that this version is Objective-C?

File: Document/0x08-Testing-Tools.md [hunk 6]
```diff
@@ -6,166 +6,163 @@ To perform security testing different tools are available in order to be able to
...
-- [DVIA](https://github.com/prateek147/DVIA "Damn Vulnerable iOS App") - A vulnerable iOS app, written in Objective-C with a set of vulnerabilities. Additional lessons can be found at [the projects website](http://damnvulnerableiosapp.com/ "DVIA project website").
-- [DVIA-v2](https://github.com/prateek147/DVIA-v2 "Damn Vulnerable iOS App v2") - A vulnerable iOS app, written in Swift with over 15 vulnerabilities.
-- [iGoat](https://github.com/owasp/igoat "iGoat") - iGoat is a learning tool for iOS developers (iPhone, iPad, etc.) and mobile app pentesters. It was inspired by the WebGoat project, and has a similar conceptual flow to it.
+ - Crackmes: A set of applications to test your iOS application hacking skills - https://github.com/OWASP/owasp-mstg/tree/master/Crackmes
+ - Myriam: A vulnerable iOS app with iOS security challenges - https://github.com/GeoSn0w/Myriam
+ - DVIA: A vulnerable iOS app which provides a platform to mobile security enthusiasts/professionals or students to test their iOS penetration testing skills - http://damnvulnerableiosapp.com/
```
Comment (sushi2k) [inline Document/0x08-Testing-Tools.md]: Can you please say that this one is in Objective-C?

File: Document/0x08-Testing-Tools.md (hunk 3 above)
Comment (sushi2k) [reply Document/0x08-Testing-Tools.md]: Please also rewrite it to address the english language errors. Then we can also close this https://github.com/OWASP/owasp-mstg/pull/1106

File: Document/0x08-Testing-Tools.md [hunk 7]
```diff
@@ -6,166 +6,163 @@ To perform security testing different tools are available in order to be able to
...
-- [Mallory](https://github.com/intrepidusgroup/mallory "Mallory") - A Man in The Middle Tool (MiTM)) that is used to monitor and manipulate traffic on mobile devices and applications.
-- [MITM Relay](https://github.com/jrmdev/mitm_relay "MITM Relay") -
-Intercept and modify non-HTTP protocols through Burp and others with support for SSL and STARTTLS interception
-- [Tcpdump](https://www.tcpdump.org/ "TCPDump") - A command line packet capture utility.
-- [Wireshark](https://www.wireshark.org/download.html "WireShark") - An open-source packet analyzer.
+ - Canape: A network testing tool for arbitrary protocols - https://github.com/ctxis/canape
+ - Mallory: A Man in The Middle Tool (MiTM) that is used to monitor and manipulate traffic on mobile devices and applications - https://github.com/intrepidusgroup/mallory
+ - MITM Relay: A script to intercept and modify non-HTTP protocols through Burp and others with support for SSL and STARTTLS interception - https://github.com/jrmdev/mitm_relay
+ - tcpdump: A command line packet capture utility - https://www.tcpdump.org/
+ - Wireshark: An open-source packet analyzer - https://www.wireshark.org/download.html
```
Comment (commjoen) [inline Document/0x08-Testing-Tools.md]: please add bettercap

Comment (cpholguera) [reply Document/0x08-Testing-Tools.md]: done

**Pull request merged**